cyflash/bootload.py

```diff
@@ -187,12 +189,24 @@ def write_rows(self, data):
 		for array_id, array in data.arrays.iteritems():
 			for row_number, row in array.iteritems():
 				i += 1
-				self.session.program_row(array_id, row_number, row.data)
-				actual_checksum = self.session.get_row_checksum(array_id, row_number)
-				if actual_checksum != row.checksum:
-					raise BootloaderError(
-						"Checksum does not match in array %d row %d. Expected %.2x, got %.2x! Aborting." % (
-							array_id, row_number, row.checksum, actual_checksum))
+                                if (data.checksum_type == 1):
+                                        self.session.program_row(array_id, row_number, row.data)
+                                elif (data.checksum_type == 0):
+                                        TRANSFER_HEADER_SIZE = 11
```

Comment:
With multiple checksum algorithms supported, the code to calculate the specific checksum should probably be handled by separate functions, with a function reference passed in to check the checksum on each block.